pkg/daemon/update.go

```diff
@@ -1228,6 +1228,11 @@ func (dn *Daemon) updateFiles(oldConfig, newConfig *mcfgv1.MachineConfig) error
 	if err := dn.writeUnits(newIgnConfig.Systemd.Units); err != nil {
 		return err
 	}
+	// In OKD /var/home/core/.ssh/authorized_keys is not created during ignition.  Create
+	// from /var/home/core/.ssh/authorized_keys.d/ignition if missing
+	if err := dn.writeMissingAuthorizedKeys(); err != nil {
```

Comment:
If this only impacts OKD, then I think it is better to add additional checks to run this only on FCOS nodes.

Reply:
Agree with @sinnykumari :100:

Reply:
This will only run on OKD nodes as OCP doesn't use /var/home/core/.ssh/authorized_keys.d/ 

that being said, is there a method to test OCP vs. OKD?

Reply:
CI should be creating test payloads for OCP and OKD.
See `build-log.txt` in `okd-e2e-aws` and `e2e-aws`, and it should have a line like `Create release image registry.build01.ci.openshift.org/ci-op-hs9i1yr4/release:latest` - this is the payload w/ test change.

However you might not have a pull secret to fetch it, so I'll help with testing those